src/OrchardCore/OrchardCore.Navigation.Core/NavigationHelper.cs

```diff
@@ -129,7 +130,7 @@ private static void ApplySelection(dynamic parentShape, ViewContext viewContext)
             // Apply the selection to the hierarchy
             if (selectedItem != null)
             {
-                viewContext.HttpContext.Response.Cookies.Append(selectedItem.Menu.MenuName + '_' + ShellScope.Context.Settings.Name, selectedItem.Hash);
+                viewContext.HttpContext.Response.Cookies.Append(selectedItem.Menu.MenuName + '_' + HttpUtility.UrlEncode(ShellScope.Context.Settings.Name), selectedItem.Hash);
```

Comment:
Please include the MenuName in the UrlEncode
```suggestion
                viewContext.HttpContext.Response.Cookies.Append(HttpUtility.UrlEncode($"{selectedItem.Menu.MenuName}_{ShellScope.Context.Settings.Name}"), selectedItem.Hash);
```